ui/raidboss/emulator/ui/EmulatedPartyInfo.ts

```diff
@@ -50,192 +139,214 @@ export default class EmulatedPartyInfo extends EventBus {
+      for (const trigger of perspective.triggers) {
         if (!trigger.status.executed || trigger.resolvedOffset > encounter.encounter.duration)
           continue;
 
-        const $e = this.$triggerItemTemplate.cloneNode(true);
-        $e.style.left = ((trigger.resolvedOffset / encounter.encounter.duration) * 100) + '%';
-        this.tooltips.push(new Tooltip($e, 'bottom', trigger.triggerHelper.trigger.id));
-        this.triggerBars[i].append($e);
+        const $e = cloneSafe(this.$triggerItemTemplate);
+        $e.style.left = ((trigger.resolvedOffset / encounter.encounter.duration) * 100).toString() + '%';
+        const triggerId = trigger.triggerHelper.trigger.id ?? 'Unknown Trigger';
+        this.tooltips.push(new Tooltip($e, 'bottom', triggerId));
+        bar.append($e);
       }
     }
 
     this.updateTriggerState();
 
-    this.selectPerspective(membersToDisplay[0]);
+    const toDisplay = membersToDisplay[0];
+    if (!toDisplay)
+      throw new UnreachableCode();
+
+    this.selectPerspective(toDisplay);
   }
 
-  selectPerspective(id) {
+  selectPerspective(id: string): void {
     if (id === this.currentPerspective)
       return;
 
-    if (!this.emulator.currentEncounter.encounter.combatantTracker.combatants[id].job)
+    if (!this.emulator.currentEncounter?.encounter.combatantTracker?.combatants[id]?.job)
       return;
 
+    const display = this.displayedParty[id];
+
+    if (!display)
+      throw new UnreachableCode();
+
     this.currentPerspective = id;
     this.$triggerInfo.querySelectorAll('.playerTriggerInfo').forEach((r) => r.classList.add('d-none'));
-    this.displayedParty[id].$triggerElem.classList.remove('d-none');
+    display.$triggerElem.classList.remove('d-none');
     this.$partyInfo.querySelectorAll('.playerInfoRow').forEach((r) => {
       r.classList.remove('border');
       r.classList.remove('border-success');
     });
-    this.displayedParty[id].$rootElem.classList.add('border');
-    this.displayedParty[id].$rootElem.classList.add('border-success');
-    this.dispatch('selectPerspective', id);
+    display.$rootElem.classList.add('border');
+    display.$rootElem.classList.add('border-success');
+    void this.dispatch('selectPerspective', id);
   }
 
-  updateCombatantInfo(encounter, id, stateID = null) {
-    if (stateID <= this.latestDisplayedState)
+  updateCombatantInfo(encounter: AnalyzedEncounter, id: string, stateID = 0): void {
+    if (!stateID || stateID <= this.latestDisplayedState)
       return;
 
-    const combatant = encounter.encounter.combatantTracker.combatants[id];
+    const combatant = encounter.encounter.combatantTracker?.combatants[id];
     if (!combatant)
       return;
-    stateID = stateID || combatant.getState(0);
 
     const State = combatant.getState(stateID);
     if (State === undefined)
       return;
 
+    const display = this.displayedParty[id];
+
+    if (!display)
+      throw new UnreachableCode();
```

Comment:
Why is this `throw new UnreachableCode` but all the other exceptions above are `return`? (Also, if you're going to assign and then check a variable, I find it easier to read with no newline between like you do above.)

Reply:
The first `return;` was correct, the rest should have been `UnreachableCode`. This did lead me to find a pointless call to `updateCombatantInfo`, cleaned up in latest commit.